### ISSUE TITLE
Revision on the Ucontrol Unbind Change

### ISSUE THREAD

**Issue by SomeonesShade** [Anuken/Mindustry-Suggestions#4631]
### Describe what you would like changed, and why.

https://github.com/Anuken/Mindustry/commit/c45f3d2253bb585ca59286dc333e2a800e080b0d
This breaks existing logic, 
there isn't really any meaningful argument for either case (before change: you would just skip through the code with a jump, or bind a new unit immediately) (after change: you would just hold on to a variable reference on the unit, ucontrol unbind, then ubind again)

It really is arbitrary either way, but just breaking existing code is a bad annoyance in general.
Before 146 comes in, i feel like this shouldn't have been changed in the first place. As unknown players may get annoyed that their logic may not work, and now need to create new changes to the code.

I feel like the benefit of changing the behavior outweighs the hidden frustration it may generate.

### Describe the changes you want to propose. Include possible alternatives.

1: Revert the commit, changing back to the original behavior.
Easiest solution back to status quo.

2: the number 1 suggestion + change the name of unbind to uncontrol
I do see it feels more misleading for something called unbind that doesn't remove the unit reference, this feels more in line on what it actually does.

*Caveat: would require making the code made in text imported as ucontrol unbind to ucontrol uncontrol

3: Create two commands, unbind (current behavior) and uncontrol (previous behavior)
Best of both worlds, having choices.

*Caveat: the frustration still remains initially, as players would now have to manually change the unbind into uncontrol.
*ucontrol has a large number of instructions, adding more may add more clutter

**Comment by Anuken** [comment]
@strangeprogrammer ?

**Comment by SomeonesShade** [comment]
Might be fairly inactive

**Comment by nanmenyangde** [comment]
or make @unit writable, and only update it when use ubind?

**Comment by strangeprogrammer** [comment]
Yes, I am fairly inactive, and also don't have much experience with GitHub's interface.  

I, personally, think that the current naming is confusing, and advocate for option 3 since it would add a little more functionality, and make old code easily fixable with find/replace in a text editor.  

Barring that, the suggestion made by 'nanmenyangde'.  

As I said in the original ticket, though, I already have a future-proof way to get the functionality I want, though it is an ugly hack (although there is additionally a much simpler method that may not be future-proof).  The reason I'm not openly sharing these techniques is because of their ugliness, which is something that no decent programming language should support, and I do not want anyone replicating them or relying upon them.

@SomeonesShade I fear that the general attitude of language programmers nowadays is to avoid simple changes that can easily be rectified by the language users, out of an abnormally large fear of backlash.  This ticket's request is in contrast to, say, removing the concept of binding altogether and feeding in a unit reference to every 'ucontrol' command, which would allow multiple units to be easily controlled at the same time by a processor, but would be noticably incompatible with the current 'mlog' scheme (in addition to possibly creating game balance issues).

I know what my own ideal programming language looks like, and 'mlog' isn't remotely close to it.  I don't intend to make 'mlog' what I want from a programming language, since a) it is already to far gone for that, and b) it's a programming language for a video game - it isn't fundamentally important to any of us in the real world (at least, I would hope) how exactly it works.

Basically, I wouldn't be making this request if I believed it could seriously break things or upset the userbase.
